src/veriflow/flows.py

```diff
@@ -16,14 +16,13 @@
 )
 from src.veriflow.networks import ConvNet2D, ConditionalDenseNN
 
-
 class Flow(torch.nn.Module):
     """Base implementation of a flow model"""
 
     # Export mode determines whether the log_prob or the sample function is exported to onnx
     export_modes = Literal["log_prob", "sample"]
     export: export_modes = "log_prob"
-    device = "cpu"
+    device = "cuda"  
```

Comment:
Is this field "device" used somewhere?

Reply:
The class variable acts as a default value in case the someone accesses the device before calling to
```python
obj = Flow(...)
obj.device
```
Since strings are immutable, the class variable cannot be changed and and instance variable "device" is created when someone sets the device variable. As soon as that happens the instance variable has precedence over the class variable.

In case that you are not familiar with class variables, have a look into [this post](https://www.toptal.com/python/python-class-attributes-an-overly-thorough-guide)

Anyway, this assumes that the initial location is always cpu. It is probably better to have it as another instantiation argument with default value "cpu".  Then you can change the value in the config file.

Reply:
Thanks for the explanation. I was wondering indeed why then we need to pass it as an argument of the fit method. I get now that is the variable we set also in the config when creating the Flow object, if I am not mistaken.